### DISCUSSION TITLE
Sync face recognition with people

### DISCUSSION THREAD

**Discussion by mizzyc** [ente/ente#803] [Enhancements]
Pretty much what Google Photos already does: enabling you to sync the face recognized with a contact (from your phone's contacts or just creating a "People" in the app).

Also, I think it's probably already planned, but I'd be nice to also have face recognition in the Android mobile app.

**Comment by laurenspriem** [comment]
Face recognition in the mobile app is indeed planned, and should be released soon! We already have a working beta, currently working on ironing out some issues and polishing some areas before we release it. 

Now for your first feature request, I don't exactly follow what you're asking for. Do you want to automatically share all photos that are recognized to contain a certain person, to share with said person? I feel I'm misunderstanding your request...

**Comment by mizzyc** [comment]
Sorry if I wasn't clear enough!
My suggestion is to improve the management of the faces that had been recognized, like Google Photos does.

For example, making me able to name a certain face as "John", so it'll be easier to search for pics with him. 
Additionally, I could be able to let Ente know that a determined face is not from "Clark" and change it to "John".

It's clearer now?

**Comment by laurenspriem** [comment]
Perfectly clear, thanks a lot! All of those things are already in the internal beta, and will be released soon ;)

**Comment by TommySharpNZ** [comment]
Can you share any detail on how you will store the face tagging? I'm so tired of losing all my face tags when I move from one photo management solution to another....

**Reply by henrivdm** [reply]
Agreed, they need to the ability to read existing face tags and to write tags into photo metadata for transferrability

**Comment by d-rk** [comment]
I also have two use-cases related to this:

Currently, the face recognition is local only. So if I have a huge shared album and one member puts the effort in to tag all persons, the other members don't benefit from this. It would be great if the tagged persons could also be shared.

> Now for your first feature request, I don't exactly follow what you're asking for. Do you want to automatically share all photos that are recognized to contain a certain person, to share with said person? I feel I'm misunderstanding your request...

I'm also interested in this feature: In my usecase we share photos taken with multiple mobile devices. This works great and photos are instantly available on all devices, but sometimes we take photos of documents or things and these photos are not worth sharing. With the restriction to share only photos of certain persons, this could be optimized.

**Reply by laurenspriem** [reply]
Thank you for your feedback!

And your second feature request is already here: https://ente.io/help/photos/features/albums-and-organization/auto-add-people

**Reply by henrivdm** [reply]
Loving Ente - but there are a couple of non-starters preventing me from jumping from Google Photos and OneDrive:
(1) Ente not reading existing face tags (and I suspect other (IPTC/XMP/Exif) tags) and having the ability to write tags into files - not as a sidecar
(2) no proper 2-way synch
Solve those, and I think many people would make the jump: Cloud Photo features to rival Google Photos, with local synch and storage of Onedrive.

**Comment by vindexins** [comment]
Problem
I've spent a lot of time grouping faces and naming people in Ente Photos. Face labels are tied to my account only. The docs confirm they aren't shared between Family plan members. Every relative has to redo the same work from scratch, which most of them (especially older relatives) won't do. So the family gets little benefit from the effort one person put in.

Proposed solution
Let me optionally share my named people with people I already share with (Family plan members, or recipients of a shared album):

- On a person's page, an option like "Share this person" or "Share names with family".
- The recipient sees an invitation and can accept or ignore it, and optionally rename it.
- If accepted, the shared label is matched to faces in their own library, or shown on photos they receive from me, so they can search by name without labeling everything again.
- Recipients can keep, rename, or remove the shared labels in their own account, and their changes never affect mine.

Privacy
This should stay opt-in and per person or per group. Labels can be shared encrypted, the same way album keys are, so the end-to-end encryption model is unchanged. Nothing is shared unless I choose to share it.

Current workaround
I can create a smart album per person (Auto-add people) and share it. That helps, but recipients still don't get a People section with names, only separate albums. Smart albums also only include photos I upload, and setup is mobile-only.

Why it matters
Family plans are meant for households and extended families. Sharing the labeling work would make face recognition much more useful for the whole family, especially for less tech-savvy members. A related idea, syncing people across accounts, has come up in the community before.

**Reply by laurenspriem** [reply]
Thank you for your feedback! We're aware of this use case, and of the problems currently surrounding it. Our first approach will be to improve the grouping out of the box, so your family will have less of a need to do the manual work. But after that, we're also planning to allow for some kind of people sync between family albums, like you suggested.

**Comment by laurenspriem** [comment]
Closing this discussion since the original request has long been completed. If there are any other feature requests in this thread that you think should be tracked then please create a separate discussion for it.
